src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,101 @@
+            : base(configuration, logger)
+        {
+        }
+
+        /// <summary>
+        ///     Path of the entity to process
+        /// </summary>
+        public string EntityPath { get; private set; }
+
+        /// <summary>
+        ///     Service Bus namespace that contains the entity
+        /// </summary>
+        public string Namespace { get; private set; }
+
+        /// <inheritdoc/>
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            Logger.LogInformation("Creating message pump");
+            var messageReceiver = CreateMessageReceiver();
+            Logger.LogInformation("Starting message pump on entity path {EntityPath} in namespace {Namespace}", EntityPath, Namespace);
+
+            // TODO: Message pump options to not delete for example
+            var messageHandlerOptions = new MessageHandlerOptions(HandleReceivedException);
+            messageReceiver.RegisterMessageHandler(HandleMessage, messageHandlerOptions);
+            Logger.LogInformation("Message pump started");
+
+            await UntilCancelledAsync(stoppingToken);
+
+            Logger.LogInformation("Closing message pump");
+            await messageReceiver.CloseAsync();
+            Logger.LogInformation("Message pump closed : {Time}", DateTimeOffset.UtcNow);
+        }
+
+        private MessageReceiver CreateMessageReceiver()
+        {
+            var connectionString = Configuration.GetValue<string>("ARCUS_SERVICEBUS_QUEUE_CONNECTIONSTRING");
+
+            var serviceBusConnectionStringBuilder = new ServiceBusConnectionStringBuilder(connectionString);
+
+            var messageReceiver = new MessageReceiver(serviceBusConnectionStringBuilder);
+            
+            EntityPath = serviceBusConnectionStringBuilder.EntityPath;
+            Namespace = messageReceiver.ServiceBusConnection.Endpoint.Host;
+
+            return messageReceiver;
+        }
+
+        private async Task HandleReceivedException(ExceptionReceivedEventArgs exceptionEvent)
+        {
+            await HandleReceiveExceptionAsync(exceptionEvent.Exception);
+        }
+
+        private async Task HandleMessage(Message message, CancellationToken cancellationToken)
+        {
+            var correlationInfo = new MessageCorrelationInfo(message.GetTransactionId(), message.CorrelationId);
+            var messageContext =
+                new AzureServiceBusMessageContext(message.MessageId, message.SystemProperties, message.UserProperties);
+
+            Logger.LogInformation("Received message '{MessageId}' (Transaction: {TransactionId}, Operation: {OperationId}, Cycle: {CycleId})",
+                messageContext.MessageId, correlationInfo.TransactionId, correlationInfo.OperationId,
+                correlationInfo.CycleId);
+            
+            // Deserialize the message
+            var typedMessageBody = DeserializeJsonMessageBody(message.Body, messageContext);
```

Comment:
don't use var + why do we assume the body will be json?

Reply:
That's what we consider to be in scope for now, more to be supported later on.